### DISCUSSION TITLE
Error using WNC string to set custom node NIP47

### DISCUSSION THREAD

**Discussion by decentralizedb** [btcpayserver/btcpayserver#6505] [Technical Support]
I'm trying to connect a store to a LN node, CLN. 
I'm using NWC. I'm generating the connection with the CLN-NWC plugin from CLN.
When I set the string to validate the connection in the BTCPay, I get the following message:
"The network of the wallet (bitcoin) does not match the network of the server (Mainnet)"

BTCPay is an advanced installation, outside of docker or Lunanode (all components installed manually).
I tested the connection using GetAlby, and it works without a problem.
On logs, I can't see any message related to this error.

Any help?

BTCPayserver: v2.0.4
CLN: v24.11
cln_nwc: [v0.1.2](https://github.com/gudnuf/cln_nwc/releases/tag/v0.1.2)
SO: Ubuntu 20.04

![image](https://github.com/user-attachments/assets/2ee329d2-181f-4a6f-bb86-829a94f2d932)

**Comment by Kukks** [comment]
The nwc implementation reports the network differently, as bitcoin instead
of mainnet. They have to change that.

**Comment by decentralizedb** [comment]
Ok, I'll open an issue with cln_nwc. 
I hope they don't tell me that the one who has to change and accept the network as Bitcoin is BTCPayserver xd.
I'll keep you informed.
Thanks.

**Comment by decentralizedb** [comment]
I think that the issue comes because the network configuration in CLN is set to Bitcoin instead of Mainnet, which is why the plugin registers that CLN value. I am confident that CLN will not change this value. 

How many CLN installations are currently in use? 

Would it be possible for BTCPayserver to accept both Bitcoin and Mainnet as valid values in these cases?

CLN config example: https://github.com/ElementsProject/lightning/blob/dcb6f2cb2eb92cfd51a528e1a1ef0dc0db99d91b/contrib/config-example#L28

![image](https://github.com/user-attachments/assets/08f700b7-109f-4d21-a83b-d4b3a8748ad7)
